var/spack/repos/builtin/packages/scotch/package.py

```diff
@@ -262,3 +270,33 @@ def install(self, spec, prefix):
         install_tree('lib', prefix.lib)
         install_tree('include', prefix.include)
         install_tree('man/man1', prefix.share.man.man1)
+
+    @when("@:6.9.99")
+    def cmake(self, spec, prefix):
+        self.configure()
+
+    @when("@:6.9.99")
+    def build(self, spec, prefix):
+        pass
+
+    def cmake_args(self):
+        spec = self.spec
+        args = [
+            self.define_from_variant('BUILD_LIBSCOTCHMETIS', 'metis'),
+            self.define_from_variant('INSTALL_METIS_HEADERS', 'metis'),
+            self.define_from_variant('BUILD_LIBESMUMPS', 'esmumps'),
+            self.define_from_variant('BUILD_SHARED_LIBS', 'shared')
+        ]
+
+        if '+mpi' in spec:
+            args.append('-DBUILD_PTSCOTCH=ON')
+            # TODO check if MPI_THREAD_MULTIPLE is supported?
+        else:
+            args.append('-DBUILD_PTSCOTCH=OFF')
```

Comment:
```suggestion
       args.append(self.define_from_variant('BUILD_PTSCOTCH', 'mpi')
```

Reply:
I was going to enable/disable `MPI_THREAD_MULTIPLE` in the if branch. But I don't see a simple way to check if it is supported.